### PR TITLE
Significant cleanup to CLI. Fixes #125. Fixes #126. Partially completes #127.

### LINKED ISSUE #125: Make all required CLI parameters positional
Except the connection profile file.

### LINKED ISSUE #126: Rename CLI commands to make them more fluent

### PR DESCRIPTION
**Pull request by BenBirt**

### CODE REVIEW

File: cli/index.ts [hunk 1]
```diff
@@ -8,75 +8,113 @@ import * as yargs from "yargs";
...
+  name: "credentials",
+  option: {
+    describe: "The location of the credentials JSON file to use.",
+    default: ".df-credentials.json",
+    coerce: filePath => {
+      filePath = path.resolve(filePath);
+      if (!fs.existsSync(filePath)) {
+        throw new Error(`Credentials file ${filePath} does not exist!`);
+      }
+      return filePath;
```
Comment (GJMcGowan) [inline cli/index.ts]: Maybe extract this function and use it here and in `projectDirOption`

File: cli/index.ts [hunk 2]
```diff
@@ -86,48 +124,26 @@ createYargsCli({
...
-        const assertionSchemaOverride = !!argv["assertion-schema-override"]
-          ? path.resolve(argv["assertion-schema-override"])
-          : "";
+        const projectDir = argv["project-dir"];
+        const defaultSchemaOverride = argv["default-schema"];
+        const assertionSchemaOverride = argv["assertion-schema"];
 
         compileProject(projectDir, defaultSchemaOverride, assertionSchemaOverride).then(() => {
-          if (argv.watch) {
+          if (argv["watch"]) {
```
Comment (GJMcGowan) [inline cli/index.ts]: What's the difference?

File: cli/index.ts (hunk 1 above)
Comment (BenBirt) [reply cli/index.ts]: Done!

File: cli/index.ts (hunk 2 above)
Comment (BenBirt) [reply cli/index.ts]: Oh there is none, I'm just trying to enforce some consistency when accessing args.

**Pull request merged**